genetic_algorithms/problems/magic_square/problem_model.py

```diff
@@ -0,0 +1,49 @@
+import math
+from pathlib import Path
+from genetic_algorithms.problems.base.model import Model
+from typing import Generator, List
+import random
+from genetic_algorithms.problems.magic_square.moves import SwapNumbers
+from genetic_algorithms.problems.magic_square.state import MagicSquareState
+import numpy as np
+
+
+class MagicSquareModel(Model):
+    def __init__(self, numbers: np.matrix, magic_number: int):
+        self._numbers: np.matrix = numbers
+        self._magic_number = magic_number
+        initial_solution = self._find_initial_solution()
+        super().__init__(initial_solution)
```

Comment:
Chyba coś, źle zmergowałeś, bo teraz `Model` wymaga też klasę `MoveGenerator`

![image](https://user-images.githubusercontent.com/21079319/135718954-50e90026-b2c7-46a5-9a27-888cfc5f34ea.png)


Ta klasa powinna robić to co teraz robisz w metodzie `moves_for`